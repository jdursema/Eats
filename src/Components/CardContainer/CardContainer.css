.card-container {
  display: flex;
  align-items: center;
  flex-wrap: nowrap;
  overflow-x: auto;
  justify-content: space-between; }

.loading {
  background-image: url("./WineWaiting.gif");
  background-position: cover;
  background-repeat: no-repeat;
  width: 100vw;
  height: 100vh; }
