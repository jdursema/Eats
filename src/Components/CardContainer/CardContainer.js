import React from 'react';
import { connect } from 'react-redux';
import Card from '../Card/Card'
import './CardContainer.css';
import { postAddFavorite } from '../../Actions';

const CardContainer = props => {
  const mappedCards = props.cards.map((card)=>{
    if (card) {
      return <Card 
        info= {card}
        handleAddFav = {props.handleAddFav}
        user = {props.user}
        history = {props.history}
      />;
    }
  });

  return (
    <div className='card-container'>
      {mappedCards}
    </div>
    
  );
};

const mapDispatchToProps = dispatch => ({
  handleAddFav: (cardData, user) => 
    dispatch(postAddFavorite(cardData, user))
});

const mapStateToProps = state => ({
  cards: state.cards,
  user: state.user
});

export default connect(mapStateToProps, mapDispatchToProps)(CardContainer);