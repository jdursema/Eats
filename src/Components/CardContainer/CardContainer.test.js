import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import CardContainer from './CardContainer';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('CardContainer', () => {
  it('should match the snapshot', ()=> {
    const mockFunc = jest.fn();
    const mockFunc2 = jest.fn();
    const cardArray = [
      {name: 'Scales', data: {
        location: 'Portland, ME'
      }},
      {name: 'Grace', data: {
        location: 'Portland, ME'
      }},
      {name: 'Miyakame', data: {
        location: 'Portland, ME'
      }}
    ]

    const renderedCardContainer = shallow(<CardContainer
      cards= {cardArray}
      handleAddFav = {mockFunc}
      handleDeleteFav = {mockFunc2}
      user = {{name: 'Julie', uid: '3jriero384u2'}}
      history = {[]}
      favorites={[{name: 'Duck Fat'}]}
    />);

    expect(renderedCardContainer).toMatchSnapshot();
  });
});