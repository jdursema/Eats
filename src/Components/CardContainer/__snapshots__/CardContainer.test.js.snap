// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cards={
        Array [
            Object {
              "data": Object {
                "location": "Portland, ME",
              },
              "name": "Scales",
            },
            Object {
              "data": Object {
                "location": "Portland, ME",
              },
              "name": "Grace",
            },
            Object {
              "data": Object {
                "location": "Portland, ME",
              },
              "name": "Miyakame",
            },
          ]
    }
    favorites={
        Array [
            Object {
              "name": "Duck Fat",
            },
          ]
    }
    handleAddFav={[Function]}
    handleDeleteFav={[Function]}
    history={Array []}
    user={
        Object {
            "name": "Julie",
            "uid": "3jriero384u2",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          favorited={false}
          favorites={
                    Array [
                              Object {
                                "name": "Duck Fat",
                              },
                            ]
          }
          handleAddFav={[Function]}
          handleDeleteFav={[Function]}
          history={Array []}
          info={
                    Object {
                              "data": Object {
                                "location": "Portland, ME",
                              },
                              "name": "Scales",
                            }
          }
          user={
                    Object {
                              "name": "Julie",
                              "uid": "3jriero384u2",
                            }
          }
/>,
        <Card
          favorited={false}
          favorites={
                    Array [
                              Object {
                                "name": "Duck Fat",
                              },
                            ]
          }
          handleAddFav={[Function]}
          handleDeleteFav={[Function]}
          history={Array []}
          info={
                    Object {
                              "data": Object {
                                "location": "Portland, ME",
                              },
                              "name": "Grace",
                            }
          }
          user={
                    Object {
                              "name": "Julie",
                              "uid": "3jriero384u2",
                            }
          }
/>,
        <Card
          favorited={false}
          favorites={
                    Array [
                              Object {
                                "name": "Duck Fat",
                              },
                            ]
          }
          handleAddFav={[Function]}
          handleDeleteFav={[Function]}
          history={Array []}
          info={
                    Object {
                              "data": Object {
                                "location": "Portland, ME",
                              },
                              "name": "Miyakame",
                            }
          }
          user={
                    Object {
                              "name": "Julie",
                              "uid": "3jriero384u2",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "favorited": false,
          "favorites": Array [
            Object {
              "name": "Duck Fat",
            },
          ],
          "handleAddFav": [Function],
          "handleDeleteFav": [Function],
          "history": Array [],
          "info": Object {
            "data": Object {
              "location": "Portland, ME",
            },
            "name": "Scales",
          },
          "user": Object {
            "name": "Julie",
            "uid": "3jriero384u2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "favorited": false,
          "favorites": Array [
            Object {
              "name": "Duck Fat",
            },
          ],
          "handleAddFav": [Function],
          "handleDeleteFav": [Function],
          "history": Array [],
          "info": Object {
            "data": Object {
              "location": "Portland, ME",
            },
            "name": "Grace",
          },
          "user": Object {
            "name": "Julie",
            "uid": "3jriero384u2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "favorited": false,
          "favorites": Array [
            Object {
              "name": "Duck Fat",
            },
          ],
          "handleAddFav": [Function],
          "handleDeleteFav": [Function],
          "history": Array [],
          "info": Object {
            "data": Object {
              "location": "Portland, ME",
            },
            "name": "Miyakame",
          },
          "user": Object {
            "name": "Julie",
            "uid": "3jriero384u2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            favorited={false}
            favorites={
                        Array [
                                    Object {
                                      "name": "Duck Fat",
                                    },
                                  ]
            }
            handleAddFav={[Function]}
            handleDeleteFav={[Function]}
            history={Array []}
            info={
                        Object {
                                    "data": Object {
                                      "location": "Portland, ME",
                                    },
                                    "name": "Scales",
                                  }
            }
            user={
                        Object {
                                    "name": "Julie",
                                    "uid": "3jriero384u2",
                                  }
            }
/>,
          <Card
            favorited={false}
            favorites={
                        Array [
                                    Object {
                                      "name": "Duck Fat",
                                    },
                                  ]
            }
            handleAddFav={[Function]}
            handleDeleteFav={[Function]}
            history={Array []}
            info={
                        Object {
                                    "data": Object {
                                      "location": "Portland, ME",
                                    },
                                    "name": "Grace",
                                  }
            }
            user={
                        Object {
                                    "name": "Julie",
                                    "uid": "3jriero384u2",
                                  }
            }
/>,
          <Card
            favorited={false}
            favorites={
                        Array [
                                    Object {
                                      "name": "Duck Fat",
                                    },
                                  ]
            }
            handleAddFav={[Function]}
            handleDeleteFav={[Function]}
            history={Array []}
            info={
                        Object {
                                    "data": Object {
                                      "location": "Portland, ME",
                                    },
                                    "name": "Miyakame",
                                  }
            }
            user={
                        Object {
                                    "name": "Julie",
                                    "uid": "3jriero384u2",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "favorited": false,
            "favorites": Array [
              Object {
                "name": "Duck Fat",
              },
            ],
            "handleAddFav": [Function],
            "handleDeleteFav": [Function],
            "history": Array [],
            "info": Object {
              "data": Object {
                "location": "Portland, ME",
              },
              "name": "Scales",
            },
            "user": Object {
              "name": "Julie",
              "uid": "3jriero384u2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "favorited": false,
            "favorites": Array [
              Object {
                "name": "Duck Fat",
              },
            ],
            "handleAddFav": [Function],
            "handleDeleteFav": [Function],
            "history": Array [],
            "info": Object {
              "data": Object {
                "location": "Portland, ME",
              },
              "name": "Grace",
            },
            "user": Object {
              "name": "Julie",
              "uid": "3jriero384u2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "favorited": false,
            "favorites": Array [
              Object {
                "name": "Duck Fat",
              },
            ],
            "handleAddFav": [Function],
            "handleDeleteFav": [Function],
            "history": Array [],
            "info": Object {
              "data": Object {
                "location": "Portland, ME",
              },
              "name": "Miyakame",
            },
            "user": Object {
              "name": "Julie",
              "uid": "3jriero384u2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
